assets/js/theme/common/carousel/index.js

```diff
@@ -2,62 +2,61 @@ import 'slick-carousel';
 
 import {
     activatePlayPauseButton,
+    analizeSlides,
     arrowAriaLabling,
     dotsSetup,
     getActiveSlideIdxAndSlidesQuantity,
     handleImageAspectRatio,
     handleImageLoad,
-    setTabindexes,
-    tooltipSetup,
+    refreshFocus,
     updateTextWithLiveData,
 } from './utils';
 
-/**
- * returns activeSlideIdx and slidesQuantity
- * based on provided carousel settings
- * @param {Object} $slickSettings
- * @returns {Object}
- */
-const extractSlidesDetails = ({
-    slideCount, $slides, options: { slidesToShow, slidesToScroll },
-}) => getActiveSlideIdxAndSlidesQuantity(
-    slideCount,
-    slidesToShow,
-    slidesToScroll,
-    $slides,
-);
+export const setCarouselState = ({ delegateTarget }, carouselObj) => {
+    const carouselObjCurrent = carouselObj || delegateTarget.slick;
+    const { $slider } = carouselObjCurrent;
+
+    $slider.data('state', getActiveSlideIdxAndSlidesQuantity(carouselObjCurrent));
+};
 
 export const onUserCarouselChange = ({ data }, context, $slider) => {
     const $activeSlider = $slider || data;
-    const $parentContainer = $activeSlider.hasClass('productView-thumbnails') ? $('.productView-images') : $activeSlider;
-    const { activeSlideIdx, slidesQuantity } = extractSlidesDetails($activeSlider[0].slick);
+    const $parentContainer = $activeSlider.hasClass('productView-thumbnails') ? $activeSlider.parent('.productView-images') : $activeSlider;
+    const { activeSlideIdx, slidesQuantity } = $activeSlider.data('state');
     const $carouselContentElement = $('[data-carousel-content-change-message]', $parentContainer);
     const carouselContentAnnounceMessage = updateTextWithLiveData(context.carouselContentAnnounceMessage, (activeSlideIdx + 1), slidesQuantity);
 
     $carouselContentElement.text(carouselContentAnnounceMessage);
 };
 
-export const onSlickCarouselChange = (e, carousel, context) => {
+export const onSlickCarouselChange = (e, carouselObj, context) => {
     const {
         $dots,
         $slider,
         $prevArrow,
         $nextArrow,
-    } = carousel;
+        options: { infinite },
+    } = carouselObj;
 
-    const { activeSlideIdx, slidesQuantity } = extractSlidesDetails(carousel);
+    const { activeSlideIdx, slidesQuantity } = $slider.data('state') || getActiveSlideIdxAndSlidesQuantity(carouselObj);
```

Comment:
Could we shorthand name of this func to getSlidesMetaData f.e?

Reply:
I would like to leave it as now, it was named in such way specially. If you will split function name you will see vars that it returns